### DISCUSSION TITLE
Custom Schema Directives with GraphQL Java SpringBoot

### DISCUSSION THREAD

**Discussion by Miraclecodes** [graphql-java/graphql-java#2272] [Q&A]
Hi, I'm trying to get Custom Schema Directives to work with my GraphQL SpringBoot Server but the directives don't seem to be adding to my schema, appearing in the playground or executing at runtime? I can't find any clear examples of how to get this working with an SDL schema and GraphQl SpringBoot.
I'm trying to implement this example but nothing is happening?
https://github.com/graphql-java/graphql-java/blob/master/src/test/groovy/readme/DirectivesExamples.java
SDL Schema Definition

`directive @auth(role : String!) on FIELD_DEFINITION
type BankAccount {
    id: ID!
    client: Client
    currency: Currency!
balance: Float @auth(role : "manager")
}
AuthorisationDirective class
@Component
public class Directives {
static class AuthorisationCtx {
    boolean hasRole(String roleName) {
        return false;
    }
    static AuthorisationCtx obtain() {
        return null;
    }
}
class AuthorisationDirective implements SchemaDirectiveWiring {
    @Override
    public GraphQLFieldDefinition onField(SchemaDirectiveWiringEnvironment<GraphQLFieldDefinition> environment) {
        String targetAuthRole = (String) environment.getDirective().getArgument("role").getValue();
        //
        // build a data fetcher that first checks authorisation roles before then calling the original data fetcher
        //
        DataFetcher originalDataFetcher = environment.getFieldDataFetcher();
        DataFetcher authDataFetcher = new DataFetcher() {
            @Override
            public Object get(DataFetchingEnvironment dataFetchingEnvironment) throws Exception {
                Map<String, Object> contextMap = dataFetchingEnvironment.getContext();
                AuthorisationCtx authContext = (AuthorisationCtx) contextMap.get("authContext");
                if (authContext.hasRole(targetAuthRole)) {
                    return originalDataFetcher.get(dataFetchingEnvironment);
                } else {
                    return null;
                }
            }
        };
        //
        // now change the field definition to have the new authorising data fetcher
        return environment.setFieldDataFetcher(authDataFetcher);
    }
}
void authWiring() {
    //
    // we wire this into the runtime by directive name
    //
    RuntimeWiring.newRuntimeWiring()
            .directive("auth", new AuthorisationDirective())
            .build();
}
String query = "";
void contextWiring() {
    AuthorisationCtx authCtx = AuthorisationCtx.obtain();
    ExecutionInput executionInput = ExecutionInput.newExecutionInput()
            .query(query)
            .context(authCtx)
            .build();
    }
}`

**Comment by gunjan-kumar-mehta** [comment]
Did you find any solution to the above problem @Miraclecodes ?
